lib/netext/httpext/httpdebug_transport.go

```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2019 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package httpext
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type httpDebugTransport struct {
+	//TODO: get the state and log to its Logger
```

Comment:
I think this should be don in the debug* methods and I would really like a test for this, as it will be very bad if we break .. the http when you are debugging for example ;)

Reply:
Not sure I understand... Can you elaborate?

Reply:
I don't want to waste any time in writing of tests that would be obsolete by the time we properly fix the `http-debug` functionality, since it's currently a pile of :hankey: . Much like the digest authentication, I haven't changed anything significant from before, I've only slightly rearranged and moved the pile of :hankey: to a separate place so it handles all cases and is easier to properly fix in the future. But I don't want to test something whose mode of working we plan to significantly change in the near future...

Reply:
I am/was left with the impression you intend on setting the state in the struct .. while I think it is okay to always get it from the request's context. 
And pointing out that a bug in this code will result in the k6/http code breaking when you try to debug it which makes it important to be tested with at least a couple of tests ;) For which being able to change the log will be useful ;)

Reply:
Being a pile of shit is the exact reason why tests are good idea ... if it wasn't it would've had less reason for it. 

Also I don't think the change will make the tests obsolete if anything the tests will show and document the change when it happens and will mostly do with (probably) small change in what we output and some caching ... both of which will not be significant problems , unlike if this code breaks or is broken in a way that nobody tested for. 

Lastly "near" and "soon" are relative terms. If you say that this will be fixed in vX.Y.Z in two weeks .. I might be fine .. but given how it usually goes I much prefer if we have tests for things that we intend to change "soon" as "soon" is usually in few months ..

Reply:
... I respectfully disagree with both points. I see no point if getting the state from the requests' context every time, since we already have it and we can just inject in the struct - much simpler, no contexts and no type asserts required.

But that point is completely moot, since I have no intentions whatsoever of dealing with any of that in this pull request, given the fact that we agreed to leave the proper fixes of the `http-debug` mess for another time... I won't dump whole requetss the logger, because I have no idea what issues that will bring - we'll deal with that properly when we fix the http-debug properly. This is just a preliminary step so it tracks every request, so that we don't make things even worse...

Reply:
FFS, now codecov complains :man_facepalming:

Reply:
This is as far as I'm willing to invest in testing HttpDebug for now: https://github.com/loadimpact/k6/pull/1102/commits/411be0cace3dd2c5fa72087bfebc0c4e039d1421